iceoryx_hoofs/platform/win/source/pthread.cpp

```diff
@@ -73,58 +78,80 @@ int pthread_mutexattr_setprotocol(pthread_mutexattr_t* attr, int protocol)
 
 int pthread_mutex_destroy(pthread_mutex_t* mutex)
 {
-    Win32Call(CloseHandle, mutex->handle);
+    if (!mutex->isInterprocessMutex)
+    {
+        Win32Call(CloseHandle, mutex->handle);
+    }
+
     return 0;
 }
 
 int pthread_mutex_init(pthread_mutex_t* mutex, const pthread_mutexattr_t* attr)
 {
-    mutex->handle =
-        Win32Call(
-            CreateMutexA, static_cast<LPSECURITY_ATTRIBUTES>(NULL), static_cast<BOOL>(FALSE), static_cast<LPCSTR>(NULL))
-            .value;
-    if (mutex->handle == NULL)
+    mutex->isInterprocessMutex = (attr != NULL && attr->isInterprocessMutex);
+
+    if (!mutex->isInterprocessMutex)
     {
-        return EINVAL;
+        mutex->handle = Win32Call(CreateMutexA,
+                                  static_cast<LPSECURITY_ATTRIBUTES>(NULL),
+                                  static_cast<BOOL>(FALSE),
+                                  static_cast<LPCSTR>(NULL))
+                            .value;
+
+
+        if (mutex->handle == NULL)
+        {
+            return EINVAL;
+        }
     }
     return 0;
 }
 
 int pthread_mutex_lock(pthread_mutex_t* mutex)
 {
-    DWORD waitResult = Win32Call(WaitForSingleObject, mutex->handle, INFINITE).value;
-
-    switch (waitResult)
+    if (!mutex->isInterprocessMutex)
```

Comment:
Is it expected that this will be used as inter-process mutex? If yes, how about printing an error to the console to inform the user instead of silently failing?

Reply:
You are right and I already implemented a fully working interprocess mutex and semaphore for windows in the upcoming PR #834 
So I would here reject it since we are not yet supporting windows and this is fixed with the next PR which is already opened as draft.